components/renku_data_services/session/db.py

```diff
@@ -0,0 +1,297 @@
+"""Adapters for session database classes."""
+
+from __future__ import annotations
+
+from datetime import datetime, timezone
+from typing import Callable
+
+from sqlalchemy import select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+import renku_data_services.base_models as base_models
+from renku_data_services import errors
+from renku_data_services.authz.authz import IProjectAuthorizer
+from renku_data_services.authz.models import MemberQualifier, Scope
+from renku_data_services.session import apispec, models
+from renku_data_services.session import orm as schemas
+from renku_data_services.session.apispec import EnvironmentKind
+
+
+class SessionRepository:
+    """Repository for sessions."""
+
+    def __init__(self, session_maker: Callable[..., AsyncSession], project_authz: IProjectAuthorizer):
+        self.session_maker = session_maker
+        self.project_authz: IProjectAuthorizer = project_authz
+
+    async def get_environments(self) -> list[models.SessionEnvironment]:
+        """Get all session environments from the database."""
+        async with self.session_maker() as session:
+            res = await session.scalars(select(schemas.SessionEnvironmentORM))
+            environments = res.all()
+            return [e.dump() for e in environments]
+
+    async def get_environment(self, environment_id: str) -> models.SessionEnvironment:
+        """Get one session environment from the database."""
+        async with self.session_maker() as session:
+            res = await session.scalars(
+                select(schemas.SessionEnvironmentORM).where(schemas.SessionEnvironmentORM.id == environment_id)
+            )
+            environment = res.one_or_none()
+            if environment is None:
+                raise errors.MissingResourceError(
+                    message=f"Session environment with id '{environment_id}' does not exist or you do not have access to it."  # noqa: E501
+                )
+            return environment.dump()
+
+    async def insert_environment(
+        self,
+        user: base_models.APIUser,
+        new_environment: apispec.EnvironmentPost,
+    ) -> models.SessionEnvironment:
+        """Insert a new session environment."""
+        if user.id is None or not user.is_admin:
```

Comment:
Should we allow creating envs for a specific project without being an admin? If yes, then it's ok to leave it for a future PR or story. Just make sure not to lose it.

Reply:
Same goes for updating and deleting envs.